### ISSUE TITLE
volume mapping does not work properly when host path is the same

### ISSUE THREAD

**Issue by useltmann** [docker/compose#983]
when i try to create a container with a volume mapping that uses the same host path, its not working

    docker run --rm -ti --name test1 -v /var/run/docker.sock:/var/run/docker.sock -v /var/run/docker.sock:/tmp/sock -v /usr/bin/docker:/usr/bin/docker busybox /usr/bin/docker inspect --format='{{.Volumes}}' test1
gives me

    map[/tmp/sock:/run/docker.sock /usr/bin/docker:/usr/bin/docker /var/run/docker.sock:/run/docker.sock]

when i run this `fig.yml` via `fig -p fig up`:

    test2:
      image: busybox
      volumes:
        - /var/run/docker.sock:/tmp/sock
        - /var/run/docker.sock:/var/run/docker.sock
        - /usr/bin/docker:/usr/bin/docker
      command: /usr/bin/docker inspect --format={{.Volumes}} fig_test2_1
i get the following

    Creating fig_test2_1...
    Attaching to fig_test2_1
    test2_1 | map[/tmp/sock:/var/lib/docker/vfs/dir/20b2479134a353d63b754900fd9363fc704bdba3dd436dba70b96052457e7618 /usr/bin/docker:/usr/bin/docker /var/run/docker.sock:/run/docker.sock]
    fig_test2_1 exited with code 0
    Gracefully stopping... (press Ctrl+C again to force)

As you can see the first mapping with `/var/run/docker.sock` is ignored and only the latter is mapped.

**Comment by kossmac** [comment]
ran into this today, so +1

**Comment by wwadge** [comment]
Same issue on centos 7  +1

**Comment by pendexgabo** [comment]
+1 .. This my prof of concept

```
web:
  image: "xxxxxxx/web:v1.7"
  command: /start.sh
  ports:
   - "49160:80"
  environment:
   - "APP_ENV=test"
  volumes:
   - /storage/:/storage/
   - ./../logs/:/app/logs/
   - ./../logs/:/var/log/
```

docker inspect shows

```
    "Volumes": {
        "/app/logs": "/var/lib/docker/vfs/dir/0c72d1274382792cafee837d853287fbadb6631b3b098d42c67e11f7a5cc732f",
        "/storage": "/storage",
        "/var/log": "/logs"
    },
```

**Comment by elalealvaro** [comment]
+1 please fix this

**Comment by lisandrok** [comment]
I encountered this issue as well on my end, so +1 here too

**Comment by juanf** [comment]
This has been open for a while, +1 for a fix please!

**Comment by cpuguy83** [comment]
All, what's the output of your `docker version`

**Comment by pendexgabo** [comment]
If I _replicate_ the commands using just docker the behaviour is the desired one. Here is the output:

```
Client version: 1.4.1
Client API version: 1.16
Go version (client): go1.3.3
Git commit (client): 5bc2ff8/1.4.1
OS/Arch (client): linux/amd64
Server version: 1.4.1
Server API version: 1.16
Go version (server): go1.3.3
Git commit (server): 5bc2ff8/1.4.1
```

**Comment by cpuguy83** [comment]
@pendexgabo Can you test with docker 1.6?

**Comment by pendexgabo** [comment]
same result with docker 1.6...

If I run `docker run -v /logs/:/var/log -v /logs/:/app/logs/ myimage` works fine.

Seems that when compose creates a map with the volumes to mount the last one overrides the first one, producing the behaviour of running run `docker run -v /logs/:/var/log -v /app/logs/ myimage`

```
Client version: 1.6.0
Client API version: 1.18
Go version (client): go1.4.2
Git commit (client): 4749651/1.6.0
OS/Arch (client): linux/amd64
Server version: 1.6.0
Server API version: 1.18
Go version (server): go1.4.2
Git commit (server): 4749651/1.6.0
OS/Arch (server): linux/amd64
```

**Comment by cpuguy83** [comment]
hmm, I wonder what the generated create/start json looks like on the compose side.

**Comment by pendexgabo** [comment]
@cpuguy83 glad to check that..I just don't seem to find a `debug` option...I have no problem to _hack_ the script, though I need some guidance where to start from.

**Comment by cpuguy83** [comment]
This looks like a compose bug.
API call is only sending the last bind.

**Comment by cpuguy83** [comment]
Confirmed, seems that it's using a dict to store bindings: https://github.com/docker/compose/blob/master/compose/service.py#L421

**Comment by cpuguy83** [comment]
Actually, this looks like the issue is in docker-py.

**Comment by aanand** [comment]
Got a fix for this in #1544, but it's waiting on a corresponding docker-py change.
